### ISSUE TITLE
微调 glm-4v 时出现错误：RuntimeError: r.nvmlDeviceGetNvLinkRemoteDeviceType_ INTERNAL ASSERT FAILED at "../c10/cuda/driver_api.cpp":27, please report a bug to PyTorch. Can't find nvmlDeviceGetNvLinkRemoteDeviceType: /lib64/libnvidia-ml.so.1: undefined symbol: nvmlDeviceGetNvLinkRemoteDeviceType

### ISSUE THREAD

**Issue by fly-dragon211** [THUDM/GLM-4#275]
### System Info / 系統信息

OS: Ubuntu 22.04
Memory: 512GB
Python: 3.10.12 
CUDA Version: 12.2
4*A100 40G

### Information / 问题信息

- [X] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

按照 https://github.com/THUDM/GLM-4/tree/main/finetune_demo 进行 finetune 

### Expected behavior / 期待表现

    return F.linear(input, self.weight, self.bias)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: r.nvmlDeviceGetNvLinkRemoteDeviceType_ INTERNAL ASSERT FAILED at "../c10/cuda/driver_api.cpp":27, please report a bug to PyTorch. Can't find nvmlDeviceGetNvLinkRemoteDeviceType: /lib64/libnvidia-ml.so.1: undefined symbol: nvmlDeviceGetNvLinkRemoteDeviceType
Train:   0%|                                          | 0/2506 [00:07<?, ?it/s]

**Comment by zRzRzRzRzRzRzR** [comment]
显存不够，每张卡都要75G

**Comment by fly-dragon211** [comment]
> 显存不够，每张卡都要75G

看着报错是Can't find nvmlDeviceGetNvLinkRemoteDeviceType，不像是显存不够？

**Comment by zRzRzRzRzRzRzR** [comment]
你这个报错都不在Pyhton层面，估计是卡之间的问题，通过nvlink去找显存么

**Comment by fly-dragon211** [comment]
> 你这个报错都不在Pyhton层面，估计是卡之间的问题，通过nvlink去找显存么

请问 nvlink 找显存是什么意思，我应该是在单张卡上训练报错。可以设置一些参数，让显存占用别这么大吗 (40G之内)

**Comment by zRzRzRzRzRzRzR** [comment]
单卡需要75G，GLM-4V-9B这个模型没有办法TP（目前没实现）所以一张卡一定要75G

**Comment by micro-cosmos** [comment]
> > 显存不够，每张卡都要75G
> 
> 看着报错是Can't find nvmlDeviceGetNvLinkRemoteDeviceType，不像是显存不够？

Hi您好！请问Can't find nvmlDeviceGetNvLinkRemoteDeviceType的问题您有解决吗？如果解决了的话是如何解决的呢？

**Comment by yangp725** [comment]
> > > 显存不够，每张卡都要75G
> > 
> > 
> > 看着报错是Can't find nvmlDeviceGetNvLinkRemoteDeviceType，不像是显存不够？
> 
> Hi您好！请问Can't find nvmlDeviceGetNvLinkRemoteDeviceType的问题您有解决吗？如果解决了的话是如何解决的呢？

可能是显存不足造成的，调低batchsize

**Comment by SiyangJ** [comment]
> > > 显存不够，每张卡都要75G
> > 
> > 
> > 看着报错是Can't find nvmlDeviceGetNvLinkRemoteDeviceType，不像是显存不够？
> 
> Hi您好！请问Can't find nvmlDeviceGetNvLinkRemoteDeviceType的问题您有解决吗？如果解决了的话是如何解决的呢？

This issue is caused by a mismatch between CUDA driver version and pytorch version.  
**TL;DR**: upgrade **CUDA driver version >= 470.42.01** or downgrade **pytorch version <= 2.2.2**.  

**Explanation**: pytorch commit [6e1ba79b7fdf3d66db8fb69462fb502e5006e5e7][1] added the following lines to pytorch/c10/cuda/driver_api.h which was released with version 2.3.0:  

    #define C10_NVML_DRIVER_API(_)           \
      _(nvmlDeviceGetNvLinkRemoteDeviceType) \
      _(nvmlDeviceGetNvLinkRemotePciInfo_v2) \

However, **nvmlDeviceGetNvLinkRemoteDeviceType** was added to nvml in **[470.42.01][2]**, while the other needed nvml functions were added to nvml no later than 450.51. Therefore, any system with CUDA driver version [450.51, 470.42.01) and pytorch version >=2.3.0 will report error when pytorch is trying to use nvlink.


  [1]: https://github.com/pytorch/pytorch/commit/6e1ba79b7fdf3d66db8fb69462fb502e5006e5e7
  [2]: https://github.com/NVIDIA/nvidia-settings/commit/cc9ff8d991d5cbcad0d4eb26cf3eb70125b60da9

**Comment by AILuckyStar** [comment]
> 显存不够，每张卡都要75G

从别的项目来的，tql大佬，torch cuda版本反复更改解不了，最后看到你这条message，用了大显存机器直接跑通了..
请问你是怎么知道是显存的原因的呢，报错完全看不出来，各种大模型的回答也一直关注版本...

**Comment by jiang-rz** [comment]
> > 显存不够，每张卡都要75G
> 
> 从别的项目来的，tql大佬，torch cuda版本反复更改解不了，最后看到你这条message，用了大显存机器直接跑通了.. 请问你是怎么知道是显存的原因的呢，报错完全看不出来，各种大模型的回答也一直关注版本...

加一，我把batchsize调小就行了

**Comment by stg1205** [comment]
nbnb
